### PR TITLE
PHOENIX-6908 KerberosName$NoMatchingRule exception in QueryServer.Pho…

### PR DESCRIPTION
**Pull request by stoty**
…enixRemoteUserExtractor

### CODE REVIEW

**Review by joshelser — COMMENTED** [review]
I remember trying to inject the realm in "production" code was a little dicey (and apt to cause errors). I don't remember exactly why anymore though. Appending the realm in test code is just fine.

Just one question about your patch, otherwise it LGTM.

File: phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java [hunk 1]
```diff
@@ -633,6 +625,26 @@ SimpleLRUCache<String,UserGroupInformation> getCache() {
...
+   */
+  //FIXME We are probably compensating for a Jetty a bug, which should really be fixed in Jetty
+  private static String stripHostNameFromPrincipal(String remoteUserName) {
+      // realm got removed from remoteUserName in CALCITE-4152
+      // so we remove the instance name to avoid geting KerberosName$NoMatchingRule exception
+      int atSignIndex = remoteUserName.indexOf('@');
+      int separatorIndex = remoteUserName.indexOf('/');
+      if (atSignIndex == -1 && separatorIndex > 0) {
+        remoteUserName = remoteUserName.substring(0, separatorIndex);
+      }
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java:644]: The hadoop api can handle phoenix/pqs.apache.org@APACHE.ORG , It breaks on phoenix/pqs.apache.org.

[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java:644]: The API being able to handle the full perincipal is just an assumption.
I'm going to test this, and try to figure out what we should get via SPNEGO from Jetty next wek.

[the remark this answers is not in the document]
Comment (joshelser) [reply phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java:644]: > I just refactored it into a method.

Gotcha. IIRC, the default `auth_to_local` rules will treat `foo/127.0.0.1`, `foo/1.2.3.4` and `foo` all as the `foo` "remote user name". This is more important for horizontally scalable services (e.g. treat all RegionServers in an HBase cluster uniformly), but I think preserving that idea in PQS is also a good idea.

Doesn't have to be done in this PR.

> I'm not sure what format principals we should support, and whether Haddop not accepting the user/host format without realm is a bug or not.

I'd imagine that the code in hadoop that can't handle the realm is considered "private API" and you would get a "we don't want to touch it" kind of response :)

Code processing principals should definitely be able to handle receiving a principal with a realm (and removing that as-necessary).

**Review by joshelser — APPROVED** [review]

**Pull request closed without merging**